### PR TITLE
SW-4520 Reports settings view and edit

### PR DESCRIPTION
**Pull request by nickgraz**
- Implement reports settings update API call
- Implement settings view and edit pages
- Create reports router

### CODE REVIEW

**Comment by nickgraz** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #2006** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2006?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2006?utm_source=stack-comment).

File: src/App.tsx [hunk 1]
```diff
@@ -528,21 +527,9 @@ function AppContent() {
...
-              </Route>
-            )}
-            {selectedOrganization.canSubmitReports && (
-              <Route path={APP_PATHS.REPORTS_VIEW}>
-                <ReportView />
-              </Route>
-            )}
+            <Route path={APP_PATHS.REPORTS}>
+              <ReportsRouter />
+            </Route>
```
Comment (karthikbtf) [inline src/App.tsx:532]: nice!

File: src/components/Reports/ReportSettingsEditForm.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  const theme = useTheme();
+  const { isMobile } = useDeviceInfo();
+  const { selectedOrganization } = useOrganization();
+  const history = useHistory();
+  const snackbar = useSnackbar();
+
+  const [localReportsSettings, setLocalReportsSettings] = useState(reportsSettings);
+
+  const onSave = useCallback(async () => {
+    const result = await ReportSettingsService.updateSettings({
```
Comment (karthikbtf) [inline src/components/Reports/ReportSettingsEditForm.tsx:27]: this is fine, wondering if using redux has any benefit here

File: src/components/Reports/ReportSettingsEditForm.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                isEnabled: value,
+              }
+        ),
+      });
+    },
+    [localReportsSettings]
+  );
+
+  return (
+    <PageForm
```
Comment (karthikbtf) [inline src/components/Reports/ReportSettingsEditForm.tsx:72]: use busy= prop?

File: src/components/Reports/ReportSettingsEditForm.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        sx={{
+          display: 'flex',
+          margin: '0 auto',
+          width: isMobile ? '100%' : '700px',
+          paddingLeft: theme.spacing(isMobile ? 0 : 4),
+          paddingRight: theme.spacing(isMobile ? 0 : 4),
+          paddingTop: theme.spacing(5),
+        }}
+      >
+        <Grid
```
Comment (karthikbtf) [inline src/components/Reports/ReportSettingsEditForm.tsx:89]: Use `Card`?

File: src/components/Reports/Router.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+const ReportsRouter = (): JSX.Element | null => {
+  const { selectedOrganization } = useOrganization();
+  const featureFlagProjects = isEnabled('Projects');
+
+  if (!selectedOrganization.canSubmitReports) {
+    return null;
+  }
+
+  return (
```
Comment (karthikbtf) [inline src/components/Reports/Router.tsx:18]: Thanks for moving this out of App!

**Review by karthikbtf — APPROVED** [review]
nice

File: src/components/Reports/ReportSettingsEditForm.tsx (hunk 2 above)
Comment (nickgraz) [reply src/components/Reports/ReportSettingsEditForm.tsx:27]: I thought about that, but it seemed like there would be a bunch of extra boilerplate for no added benefit.

Comment (karthikbtf) [reply src/components/Reports/ReportSettingsEditForm.tsx:27]: Got it, sounds fair to me.

**Pull request merged**